cpp/CMakeLists.txt

```diff
@@ -322,15 +322,31 @@ endif(PER_THREAD_DEFAULT_STREAM)
 
 if(BUILD_TESTS)
     include(CTest)
-    include(ConfigureGoogleTest)
+    find_package(GTest)
+    
+    if(GTEST_ROOT)
+        message(STATUS "Google C++ Testing Framework (Google Test) found in ${GTEST_ROOT}")
```

Comment:
Can move this status message to outside of these find / build statements to below them and replace the current status message.

Reply:
Like just carry a status variable and push the message once?

Reply:
There is no `STATUS` variable, `STATUS` is just a message type here, and payload of the message is converying `GTEST_ROOT` which is set on either finding the `gtest` package or on building it, so may as well convey the status message after both had run. If you wanted to add a different message per whether we find gtest or build it from source I think that would be great.

Reply:
A+